src/test/java/uk/gov/hmcts/reform/bulkscanprocessor/services/reports/ReconciliationServiceTest.java

```diff
@@ -0,0 +1,420 @@
+        );
+        ReconciliationStatement statement = new ReconciliationStatement(date, envelopes);
+
+        // when
+        List<Discrepancy> discrepancies = reconciliationService.getReconciliationReport(statement);
+
+        // then
+        assertThat(discrepancies)
+            .usingRecursiveFieldByFieldElementComparator()
+            .containsExactlyInAnyOrder(
+                new Discrepancy("file1", "c1", PAYMENT_DCNS_MISMATCH.text, null,"[pay-1, pay-2]")
+            );
+    }
+
+    @Test
+    void should_handle_empty_reported_payment_dcns() {
+        // given
+        LocalDate date = LocalDate.now();
+
+        final List<ReceivedZipFileData> receivedZipFileDataList = asList(
+            new ReceivedZipFileData("file1", "c1", asList("doc-1", "doc-2"), asList("pay-1", "pay-2")),
+            new ReceivedZipFileData("file2", "c2", asList("doc-3", "doc-4"), asList("pay-3", "pay-4"))
+        );
+
+        prepareReconciliationServiceBehaviour(date, receivedZipFileDataList);
+
+        List<ReportedZipFile> envelopes = asList(
+            new ReportedZipFile("file1", "c1", null, asList("doc-1", "doc-2"), emptyList()),
+            new ReportedZipFile("file2", "c2", null, asList("doc-3", "doc-4"), asList("pay-3", "pay-4"))
+        );
+        ReconciliationStatement statement = new ReconciliationStatement(date, envelopes);
+
+        // when
+        List<Discrepancy> discrepancies = reconciliationService.getReconciliationReport(statement);
+
+        // then
+        assertThat(discrepancies)
+            .usingRecursiveFieldByFieldElementComparator()
+            .containsExactlyInAnyOrder(
+                new Discrepancy("file1", "c1", PAYMENT_DCNS_MISMATCH.text, "[]","[pay-1, pay-2]")
+            );
+    }
+
+    @Test
+    void should_handle_empty_received_payment_dcns() {
+        // given
+        LocalDate date = LocalDate.now();
+
+        final List<ReceivedZipFileData> receivedZipFileDataList = asList(
+            new ReceivedZipFileData("file1", "c1", asList("doc-1", "doc-2"), emptyList()),
+            new ReceivedZipFileData("file2", "c2", asList("doc-3", "doc-4"), asList("pay-3", "pay-4"))
+        );
+        prepareReconciliationServiceBehaviour(date, receivedZipFileDataList);
+
+        List<ReportedZipFile> envelopes = asList(
+            new ReportedZipFile("file1", "c1", null, asList("doc-1", "doc-2"), asList("pay-1", "pay-2")),
+            new ReportedZipFile("file2", "c2", null, asList("doc-3", "doc-4"), asList("pay-3", "pay-4"))
+        );
+        ReconciliationStatement statement = new ReconciliationStatement(date, envelopes);
+
+        // when
+        List<Discrepancy> discrepancies = reconciliationService.getReconciliationReport(statement);
+
+        // then
+        assertThat(discrepancies)
+            .usingRecursiveFieldByFieldElementComparator()
+            .containsExactlyInAnyOrder(
+                new Discrepancy("file1", "c1", PAYMENT_DCNS_MISMATCH.text, "[pay-1, pay-2]","[]")
```

Comment:
it would be good to have at least one test that covers a case when there are more than 1 discrepancy.

Reply:
`should_return_reported_but_not_received_and_received_but_not_reported_discrepancies` - this is hopefully what does this. I will add more tests